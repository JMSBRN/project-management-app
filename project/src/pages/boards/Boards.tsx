import React, { useEffect, useState } from 'react';
import {
  Board,
  BoardsWrapper,
  BoardWrapper,
  Description,
  Icons,
  NewBoard,
  NewBoardWrapper,
  Title,
  Wrapper,
} from './Boards.style';
import editBoard from '../../assets/img/edit.png';
import deleteBoardImg from '../../assets/img/delete.png';
import ModalDelete from 'components/modalDelete/ModalDelete';
import BoardForm from 'components/boardForm/BoardForm';
import { useNavigate } from 'react-router-dom';
import { useAppDispatch, useAppSelector } from 'app/hooks';
import { setBoardsBtns } from 'features/api/ApiSlice';
import { boardsSelect, newBoardId, deleteBoard } from 'features/boards/BoardsSlice';

export interface IBoard {
  title: string;
  text: string;
  columns: IColumns[];
}

export interface IData {
  id: string;
  Task: string;
  message: string;
}

export interface IColumns {
  title: string;
  items: IData[] | [];
}

export function getRandomID() {
  return (Math.random() + 1).toString(36).substring(7);
}

const Boards = () => {
  const dispatch = useAppDispatch();
  const { boards, boardId } = useAppSelector(boardsSelect);
  const navigate = useNavigate();
  const [changeBoard, setchangeBoard] = useState(false);
  const [isDelete, setisDelete] = useState(false);

  const deleteBoards = (id: number) => {
    const newBoardsArr = boards.filter((n, index) => {
      return index !== id;
    });
    dispatch(deleteBoard(newBoardsArr));
  };

  useEffect(() => {
    dispatch(setBoardsBtns(true));
  }, [dispatch]);

  return (
    <Wrapper>
      <BoardsWrapper>
        {boards.map((item: IBoard, index: number) => {
          return (
            <BoardWrapper key={index}>
              <Board
                onClick={() => {
                  navigate('/board');
                  dispatch(newBoardId(index));
                }}
              >
                <Title>{item.title}</Title>
                <Description>{item.text}</Description>
              </Board>
              <div>
                <Icons
                  img={editBoard}
                  onClick={() => {
                    setchangeBoard(true);
                    dispatch(newBoardId(index));
                  }}
                />
                <Icons
                  img={deleteBoardImg}
                  onClick={() => {
                    setisDelete(true);
                    dispatch(newBoardId(index));
                  }}
                />
              </div>
            </BoardWrapper>
          );
        })}
        <NewBoardWrapper
          onClick={() => {
            setchangeBoard(true);
            dispatch(newBoardId(boards.length + 1));
          }}
        >
          <NewBoard />
        </NewBoardWrapper>
        {changeBoard && (
          <BoardForm setchangeBoard={setchangeBoard} boardId={boardId} boards={boards} />
        )}
        {isDelete && (
          <ModalDelete deleteBoards={deleteBoards} boardId={boardId} setisDelete={setisDelete} />
        )}
      </BoardsWrapper>
    </Wrapper>
  );
};

export default Boards;
