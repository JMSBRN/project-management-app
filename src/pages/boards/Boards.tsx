import React, { useEffect, useState } from 'react';
import {
  Board,
  BoardsWrapper,
  BoardWrapper,
  Description,
  Icons,
  NewBoard,
  NewBoardWrapper,
  Title,
  Wrapper,
} from './Boards.style';
import editBoard from '../../assets/img/edit.png';
import deleteBoard from '../../assets/img/delete.png';
import ModalDelete from 'components/modalDelete/ModalDelete';
import BoardForm from 'components/boardForm/BoardForm';
import { useNavigate } from 'react-router-dom';

export interface IBoard {
  title: string;
  text: string;
}

const Boards = () => {
  const navigate = useNavigate();
  const [boards, setBoards] = useState<IBoard[]>([
    { title: 'task', text: 'dsadfas' },
    {
      title: 'task',
      text: 'dsadfssss',
    },
    { title: 'task', text: 'dsadfasdadfawfaw' },
    { title: 'tasaSAsk', text: 'dsadfassaSAsdadfawfaw' },
    { title: 'tasaSAsk', text: 'dsadfassaSAsdadfawfaw' },
    { title: 'tasaSAsk', text: 'dsadfassaSAsdadfawfaw' },
  ]);
  const [changeBoard, setchangeBoard] = useState(false);
  const [isDelete, setisDelete] = useState(false);
  const [DeleteBoard, setDeleteBoard] = useState(false);
  const [BoardId, setBoardId] = useState<null | number>(null);

  useEffect(() => {
    const deleteBoards = (id: number) => {
      if (DeleteBoard) {
        const newBoards = boards.filter((n, index) => {
          return index !== id;
        });
        setBoards(newBoards);
      }
    };
    if (DeleteBoard) {
      deleteBoards(BoardId!);
      setBoardId(null);
      setDeleteBoard(false);
    }
  }, [DeleteBoard, BoardId, boards]);

  return (
    <Wrapper>
      <BoardsWrapper>
        {boards.map((item, index) => {
          return (
            <BoardWrapper key={index}>
              <Board onClick={() => navigate('/board')}>
                <Title>{item.title}</Title>
                <Description>{item.text}</Description>
              </Board>
              <div>
                <Icons
                  img={editBoard}
                  onClick={() => {
                    setchangeBoard(true);
                    setBoardId(index);
                  }}
                />
                <Icons
                  img={deleteBoard}
                  onClick={() => {
                    setisDelete(true);
                    setBoardId(index);
                  }}
                />
              </div>
            </BoardWrapper>
          );
        })}
        <NewBoardWrapper
          onClick={() => {
            setchangeBoard(true);
            setBoardId(boards.length + 1);
          }}
        >
          <NewBoard />
        </NewBoardWrapper>
        {changeBoard && (
          <BoardForm
            setchangeBoard={setchangeBoard}
            BoardId={BoardId}
            boards={boards}
            setBoards={setBoards}
          />
        )}
        {isDelete && <ModalDelete setisDelete={setisDelete} setDelete={setDeleteBoard} />}
      </BoardsWrapper>
    </Wrapper>
  );
};

export default Boards;
