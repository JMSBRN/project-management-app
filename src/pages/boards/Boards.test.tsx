import { render } from '@testing-library/react';
import React from 'react';
import Boards from './Boards';

describe('<Boards />', () => {
  it('should be ', () => {
    const { asFragment } = render(<Boards />);
    expect(asFragment).toMatchSnapshot();
  });
});
