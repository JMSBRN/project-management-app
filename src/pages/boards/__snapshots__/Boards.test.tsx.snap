// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Boards /> should be  1`] = `[Function]`;
